csrc/type.cpp

```diff
@@ -379,6 +379,10 @@ static const char* binary_op_type2string(BinaryOpType t) {
       return "remainder";
     case BinaryOpType::Sub:
       return "sub";
+    case BinaryOpType::ComplexFloat:
+      return "std::complex<float>";
+    case BinaryOpType::ComplexDouble:
+      return "std::complex<double>";
```

Comment:
I think we should only have a single binary op type, and we can use [the CTAD of C++17](https://en.cppreference.com/w/cpp/language/class_template_argument_deduction) to infer the returning type

Reply:
CTAD works in this case. At the beginning I used std::compex but doesn't work, maybe due to other reason.